libbeat/processors/fingerprint/hash_nofips.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build !requirefips
+
+package fingerprint
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+)
+
+var namedHashMethods = []namedHashMethod{
+	{Name: "md5", Hash: md5.New},
```

Comment:
Nit.

Could we maybe move to where we have 2 slices, one FIPS approved and one Non-FIPS approved.  Then something in a build tag that controls merging them into a slice with the correct values.  What I've seen in previous FIPS approved products, keeping 2 separate lists in 2 separate files makes it very easy for them to get out of sync.

Even better if we could have one place in libbeat to update the list of approved hashes functions.

Reply:
I was thinking of something like this:

```
// hash.go
// no build tag

var hashes = map[string]namedHashMethod{}

var fipsApprovedHashes = []namedHashMethod{
	{Name: "sha512", Hash: sha512.New},
}
```
```
//hash_fips.go
////go:build requirefips

func init() {
	for _, h := range fipsApprovedHashes {
		hashes[h.Name] = h
	}
}
```
```
//hash_nofips.go
//go:build !requirefips

var nonFipsApprovedHashes = []namedHashMethod{
	{Name: "md5", Hash: md5.New},
}

func init() {
	for _, h := range fipsApprovedHashes {
		hashes[h.Name] = h
	}
	for _, h := range nonFipsApprovedHashes {
		hashes[h.Name] = h
	}
}
```

FIPS 
hashes = fipsApprovedHashes

NonFips
hashes = fipsApprovedHashes + nonFipsApprovedHashes

Reply:
Seconding that I would also like to see a change along these lines, otherwise looks good.